Feature: Display All Products

ksookdhew opened the pull request:
<img width="380" alt="Screenshot 2024-08-07 at 08 03 56" src="https://github.com/user-attachments/assets/e38fec94-7ff1-4f9c-bf8b-122b647e8eb3">
<img width="1727" alt="Screenshot 2024-08-07 at 08 04 08" src="https://github.com/user-attachments/assets/8c4c85ab-f622-4d00-82bd-7ae0510bca6e">

ksookdhew committed:
chore: Created products model

Created the models for the product endpoint

ksookdhew committed:
feat: Created function components

Created the productCard and AllProducts (Product container) that will be used to render the home page

ksookdhew committed:
feat: fetch all products

fetch all products from the fakeStore api

ksookdhew committed:
feat: fetch categories

fetch and display the categories from the fakeStore api

sjacobs1 commented on src/features/allProducts/AllProducts.tsx:
```diff
@@ -0,0 +1,29 @@
+const AllProducts = () => {
+    const allProducts = useQuery({queryKey: ['allProducts'], queryFn: getAllProducts})
+    const categories = useQuery({queryKey: ['categories'], queryFn: getAllCategories})
+
```
just a heads up for the loading and error cases

AdeWet commented on src/features/allProducts/AllProducts.tsx:
```diff
@@ -0,0 +1,29 @@
+            <div className="carousel carousel-center w-full space-x-4 self-center">
+                {
+                    categories.data?.map((category) => (
+                        <div className="carousel-item">
```
just a reminder to use a key :)

ksookdhew committed:
fix: query cases and map key

Added loading and error cases for the api call and added key to the map function

ksookdhew committed:
fix: delete empty file

Deleted the empty file
